[receiver/sqlquery] Flaky test TestMysqlIntegration

### Component(s)

receiver/sqlquery

### What happened?

See https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/4988067592/jobs/8930453175?pr=21991

```
--- FAIL: TestMysqlIntegration (41.62s)
panic: runtime error: index out of range [0] with length 0 [recovered]
	panic: runtime error: index out of range [0] with length 0

goroutine 1128 [running]:
testing.tRunner.func1.2({0x1ca3300, 0xc000c10900})
	/opt/hostedtoolcache/go/1.19.9/x64/src/testing/testing.go:1396 +0x372
testing.tRunner.func1()
	/opt/hostedtoolcache/go/1.19.9/x64/src/testing/testing.go:1399 +0x5f0
panic({0x1ca3300, 0xc000c10900})
	/opt/hostedtoolcache/go/1.19.9/x64/src/runtime/panic.go:890 +0x262
go.opentelemetry.io/collector/pdata/pmetric.ResourceMetricsSlice.At(...)
	/home/runner/go/pkg/mod/go.opentelemetry.io/collector/pdata@v1.0.0-rcv0011/pmetric/generated_resourcemetricsslice.go:64
github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlqueryreceiver.TestMysqlIntegration(0xc000aceb60?)
	/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sqlqueryreceiver/integration_test.go:362 +0xbd0
testing.tRunner(0xc000aceb60, 0x1dfecb8)
	/opt/hostedtoolcache/go/1.19.9/x64/src/testing/testing.go:1446 +0x217
created by testing.(*T).Run
	/opt/hostedtoolcache/go/1.19.9/x64/src/testing/testing.go:1493 +0x75e
FAIL	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlqueryreceiver	175.542s
?   	github.com/open-telemetry/opentelemetry-collector-contrib/receiver/sqlqueryreceiver/internal/metadata	[no test files]
FAIL
make[1]: *** [../../Makefile.Common:109: do-integration-tests-with-cover] Error 1
make[1]: Leaving directory '/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/receiver/sqlqueryreceiver'
make: *** [Makefile:173: receiver/sqlqueryreceiver] Error 2
```

### Collector version

latest

### Environment information

## Environment
OS: (e.g., "Ubuntu 20.04")
Compiler(if manually compiled): (e.g., "go 14.2")


### OpenTelemetry Collector configuration

_No response_

### Log output

_No response_

### Additional context

_No response_

--- a/receiver/sqlqueryreceiver/integration_test.go
+++ b/receiver/sqlqueryreceiver/integration_test.go
@@ -139,13 +139,15 @@ func TestPostgresIntegration(t *testing.T) {
 	require.Eventuallyf(
 		t,
 		func() bool {
-			return consumer.DataPointCount() > 0
+			ms := consumer.AllMetrics()
+			return len(ms) > 0 && ms[len(ms)-1].ResourceMetrics().Len() >= 2
 		},
 		2*time.Minute,
 		1*time.Second,
 		"failed to receive more than 0 metrics",
 	)
-	metrics := consumer.AllMetrics()[0]
+	allMetrics := consumer.AllMetrics()
+	metrics := allMetrics[len(allMetrics)-1]
 	rms := metrics.ResourceMetrics()
 	testMovieMetrics(t, rms.At(0), genreKey)
 	testPGTypeMetrics(t, rms.At(1))
@@ -225,7 +227,8 @@ func TestOracleDBIntegration(t *testing.T) {
 		1*time.Second,
 		"failed to receive more than 0 metrics",
 	)
-	metrics := consumer.AllMetrics()[0]
+	allMetrics := consumer.AllMetrics()
+	metrics := allMetrics[len(allMetrics)-1]
 	rms := metrics.ResourceMetrics()
 	testMovieMetrics(t, rms.At(0), genreKey)
 }
@@ -340,13 +343,15 @@ func TestMysqlIntegration(t *testing.T) {
 	require.Eventuallyf(
 		t,
 		func() bool {
-			return consumer.DataPointCount() > 0
+			ms := consumer.AllMetrics()
+			return len(ms) > 0 && ms[len(ms)-1].ResourceMetrics().Len() >= 2
 		},
 		2*time.Minute,
 		1*time.Second,
 		"failed to receive more than 0 metrics",
 	)
-	metrics := consumer.AllMetrics()[0]
+	allMetrics := consumer.AllMetrics()
+	metrics := allMetrics[len(allMetrics)-1]
 	rms := metrics.ResourceMetrics()
 	testMovieMetrics(t, rms.At(0), genreKey)
 	testMysqlTypeMetrics(t, rms.At(1))

